Introduce incremental compaction strategy (ICS)

raphaelsc opened the pull request:
ICS is a compaction strategy that inherits size tiered properties --
therefore it's write optimized too -- but fixes its space overhead of
100% due to input files being only released on completion. That's
achieved with the concept of sstable run (similar in concept to LCS
levels) which breaks a large sstable into fixed-size chunks (1G by
default), known as run fragments. ICS picks similar-sized runs
for compaction, and fragments of those runs can be released
incrementally as they're compacted, reducing the space overhead
to about (number_of_input_runs * 1G). This allows user to increase
storage density of nodes (from 50% to ~80%), reducing the cost of
ownership.

NOTE: test_system_schema_version_is_stable adjusted to account for batchlog using IncrementalCompactionStrategy

contains:

```
compaction/: added incremental_compaction_strategy.cc (.hh), incremental_backlog_tracker.cc (.hh)
compaction/CMakeLists.txt: include ICS cc files
configure.py: changes for ICS files, includes test
db/legacy_schema_migrator.cc / db/schema_tables.cc: fallback to ICS when strategy is not supported
db/system_keyspace: pick ICS for some system tables
schema/schema.hh: ICS becomes default
test/boost: Add incremental_compaction_test.cc
test/boost/sstable_compaction_test.cc: ICS related changes
test/cqlpy/test_compaction_strategy_validation.py: ICS related changes

docs/architecture/compaction/compaction-strategies.rst: changes to ICS section
docs/cql/compaction.rst: changes to ICS section
docs/cql/ddl.rst: adds reference to ICS options
docs/getting-started/system-requirements.rst: updates sentence mentioning ICS
docs/kb/compaction.rst: changes to ICS section
docs/kb/garbage-collection-ics.rst: add file
docs/kb/index.rst: add reference to <garbage-collection-ics>
docs/operating-scylla/procedures/tips/production-readiness.rst: add ICS section
```

some relevant commits throughout the ICS history:


```
    Merge "Introduce Incremental Compaction Strategy (ICS)" from Raphael

    "
    Introduce new compaction strategy which is essentially like size tiered
    but will work with the existing incremental compaction. Thus incremental
    compaction strategy.

    It works like size tiered, but each element composing a tier is a sstable
    run, meaning that the compaction strategy will look for N similar-sized
    sstable runs to compact, not just individual sstables.

    Parameters:
    * "sstable_size_in_mb": defines the maximum sstable (fragment) size composing a sstable run, which impacts directly the disk space requirement which is improved with incremental compaction. The lower the value the lower the space requirement for compaction because fragments involved will be released more frequently.
    * all others available in size tiered compaction strategy

    HOWTO
    =====

    To change an existing table to use it, do:
         ALTER TABLE mykeyspace.mytable  WITH compaction =
    {'class' : 'IncrementalCompactionStrategy'};

    Set fragment size:
         ALTER TABLE mykeyspace.mytable  WITH compaction =
    {'class' : 'IncrementalCompactionStrategy', 'sstable_size_in_mb' : 1000 }

    "
```

```
    Merge "Add feature to limit space amplification in Incremental Compaction" from Raphael

    "
    A new option, space_amplification_goal (SAG), is being added to ICS. This option
    will allow ICS user to set a goal on the space amplification (SA). It's not
    supposed to be an upper bound on the space amplification, but rather, a goal.
    This new option will be disabled by default as it doesn't benefit write-only
    (no overwrites) workloads and could hurt severely the write performance.
    The strategy is free to delay triggering this new behavior, in order to
    increase overall compaction efficiency.

    The graph below shows how this feature works in practice for different values
    of space_amplification_goal:
    https://user-images.githubusercontent.com/1409139/89347544-60b7b980-d681-11ea-87ab-e2fdc3ecb9f0.png

    When strategy finds space amplification crossed space_amplification_goal, it
    will work on reducing the SA by doing a cross-tier compaction on the two
    largest tiers. This feature works only on the two largest tiers, because taking
    into account others, could hurt the compaction efficiency which is based on
    the fact that the more similar-sized sstables are compacted together the higher
    the compaction efficiency will be.

    With SAG enabled, min_threshold only plays an important role on the smallest
    tiers, given that the second-largest tier could be compacted into the largest
    tier for a space_amplification_goal value < 2.
    By making the options space_amplification_goal and min_threshold independent,
    user will be able to tune write amplification and space amplification, based on
    the needs. The lower the space_amplification_goal the higher the write
    amplification, but by increasing the min threshold, the write amplification
    can be decreased to a desired amount.
    "
```

```
    compaction: ICS: Add garbage collection

    Today, ICS lacks an approach to persist expired tombstones in a timely manner,
    which is a problem because accumulation of tombstones are known to affecting
    latency considerably.

    For an expired tombstone to be purged, it has to reach the top of the LSM tree
    and hope that older overlapping data wasn't introduced at the bottom.
    The condition are there and must be satisfied to avoid data resurrection.

    STCS, today, has an inefficient garbage collection approach because it only
    picks a single sstable, which satisfies the tombstone density threshold and
    file staleness. That's a problem because overlapping data either on same tier
    or smaller tiers will prevent tombstones from being purged. Also, nothing is
    done to push the tombstones to the top of the tree, for the conditions to be
    eventually satisfied.

    Due to incremental compaction, ICS can more easily have an effecient GC by
    doing cross-tier compaction of relevant tiers.

    The trigger will be file staleness and tombstone density, which threshold
    values can be configured by tombstone_compaction_interval and
    tombstone_threshold, respectively.

    If ICS finds a tier which meets both conditions, then that tier and the
    larger[1] *and* closest-in-size[2] tier will be compacted together.
    [1]: A larger tier is picked because we want tombstones to eventually reach the
    top of the tree.
    [2]: It also has to be the closest-in-size tier as the smaller the size
    difference the higher the efficiency of the compaction. We want to minimize
    write amplification as much as possible.
    The staleness condition is there to prevent the same file from being picked
    over and over again in a short interval.

    With this approach, ICS will be continuously working to purge garbage while
    not hurting overall efficiency on a steady state, as same-tier compactions are
    prioritized.

    Signed-off-by: Raphael S. Carvalho <raphaelsc@scylladb.com>
    Message-Id: <20211016164146.38010-1-raphaelsc@scylladb.com>
```

New feature, not eligible for backport.

scylladb-promoter commented:
## Docs Preview :book:
Docs Preview for this pull request is available [here](http://previews.docs.scylladb.com/scylladb/22063)

**Changed Files:**
- [architecture/compaction/compaction-strategies](http://previews.docs.scylladb.com/scylladb/22063/architecture/compaction/compaction-strategies)
- [cql/compaction](http://previews.docs.scylladb.com/scylladb/22063/cql/compaction)
- [cql/ddl](http://previews.docs.scylladb.com/scylladb/22063/cql/ddl)
- [getting-started/system-requirements](http://previews.docs.scylladb.com/scylladb/22063/getting-started/system-requirements)
- [kb/compaction](http://previews.docs.scylladb.com/scylladb/22063/kb/compaction)
- [kb/index.html](http://previews.docs.scylladb.com/scylladb/22063/kb/index.html)
- [operating-scylla/procedures/tips/production-readiness](http://previews.docs.scylladb.com/scylladb/22063/operating-scylla/procedures/tips/production-readiness)

**Note:** This preview will be available for 30 days and will be automatically deleted after that period. You can manually trigger a new build by committing changes.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14098/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14098/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: boost/incremental_compaction_test
:small_blue_diamond: boost/schema_change_test
:small_blue_diamond: boost/sstable_compaction_test
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5644/)
:x: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2530/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13657/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/900/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/304/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14098/)
#### Failed Tests (19/37702):
- [test_loading_cache_update_config](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/boost/loading_cache_test/test_loading_cache_update_config) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_loading_cache_update_config)
- [test_describe](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_tablets___test_describe) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe)
- [test_describe](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest___test_describe) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe)
- [test_describe](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_topology_changes___test_describe) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe)
- [test_describe_mv](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest___test_describe_mv) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe_mv)
- [test_describe_mv](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_tablets___test_describe_mv) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe_mv)
- [test_describe_mv](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_topology_changes___test_describe_mv) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe_mv)
- [test_describe[require_client_auth=false]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlshWithSSL/Tests___dtest___test_describe_require_client_auth_false_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe[require_client_auth=false])
- [test_describe[require_client_auth=false]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlshWithSSL/Tests___dtest_with_tablets___test_describe_require_client_auth_false_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe[require_client_auth=false])
- [test_describe[require_client_auth=false]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlshWithSSL/Tests___dtest_with_topology_changes___test_describe_require_client_auth_false_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe[require_client_auth=false])
#### Build Details:
- Duration: 5 hr 8 min
- Builder: spider8.cloudius-systems.com

raphaelsc commented:
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14098/) ✅ - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14098/) **The following new/updated tests ran 100 times for each mode:** 🔹 boost/incremental_compaction_test 🔹 boost/schema_change_test 🔹 boost/sstable_compaction_test ❌ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5644/) ❌ - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2530/) ❌ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13657/) ✅ - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/900/) ✅ - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/304/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14098/)
> 
> #### Failed Tests (19/37702):
> * [test_loading_cache_update_config](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/boost/loading_cache_test/test_loading_cache_update_config) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_loading_cache_update_config)
> * [test_describe](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_tablets___test_describe) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe)
> * [test_describe](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest___test_describe) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe)
> * [test_describe](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_topology_changes___test_describe) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe)
> * [test_describe_mv](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest___test_describe_mv) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe_mv)
> * [test_describe_mv](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_tablets___test_describe_mv) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe_mv)
> * [test_describe_mv](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlsh/Tests___dtest_with_topology_changes___test_describe_mv) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe_mv)
> * [test_describe[require_client_auth=false]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlshWithSSL/Tests___dtest___test_describe_require_client_auth_false_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe%5Brequire_client_auth=false%5D)
> * [test_describe[require_client_auth=false]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlshWithSSL/Tests___dtest_with_tablets___test_describe_require_client_auth_false_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe%5Brequire_client_auth=false%5D)
> * [test_describe[require_client_auth=false]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14098/testReport/junit/cqlsh_tests.cqlsh_tests/TestCqlshWithSSL/Tests___dtest_with_topology_changes___test_describe_require_client_auth_false_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_describe%5Brequire_client_auth=false%5D)
> 
> #### Build Details:
> * Duration: 5 hr 8 min
> * Builder: spider8.cloudius-systems.com

tests fail because scylla-dtest in OSS expects STCS, I suggested we patch it so it accepts either temporarily.

bhalevy commented:
> tests fail because scylla-dtest in OSS expects STCS, I suggested we patch it so it accepts either temporarily.

That's fine.
Please go ahead and submit a patch for scylla-dtest.

bhalevy commented on compaction/compaction_strategy.cc:
```diff
@@ -174,8 +177,11 @@ void compaction_strategy_impl::validate_options_for_strategy_type(const std::map
+            incremental_compaction_strategy::validate_options(options, unchecked_options);
             break;
+        case compaction_strategy_type::null:
+            return;
```
`default:` had `break` before the change.
We shouldn't change that in this patch.
And it's nit a bad idea to erase the `NullCompactionStrategy` class form the options, same as for any other strategy, even if it's not strictly requried.

bhalevy commented on docs/cql/compaction.rst:
```diff
@@ -170,7 +170,106 @@ LCS options
-ICS is only available in ScyllaDB Enterprise. See the `ScyllaDB Enetrpise documentation <https://enterprise.docs.scylladb.com/stable/cql/compaction.html>`_ for details.
+.. versionadded:: 2019.1.4 Scylla Enterprise
+
+.. note:: ICS is only available for Scylla Enterprise customers
```
outdated.

bhalevy commented on docs/architecture/compaction/compaction-strategies.rst:
```diff
@@ -70,7 +70,38 @@ Set the parameters for :ref:`Leveled Compaction <leveled-compaction-strategy-lcs
-ICS is only available in ScyllaDB Enterprise. See the `ScyllaDB Enetrpise documentation <https://enterprise.docs.scylladb.com/stable/architecture/compaction/compaction-strategies.html>`_ for details.
+.. versionadded:: 2019.1.4 Scylla Enterprise
+
+.. note:: ICS is only available for Scylla Enterprise customers
```
outdated.

bhalevy commented on docs/architecture/compaction/compaction-strategies.rst:
```diff
@@ -70,7 +70,38 @@ Set the parameters for :ref:`Leveled Compaction <leveled-compaction-strategy-lcs
+
+.. note:: ICS is only available for Scylla Enterprise customers
+
+ICS principles of operation are similar to those of STCS, merely replacing the increasingly larger SSTables in each tier, by increasingly longer SSTable runs, modeled after LCS runs, but using larger fragment size of 1 GB, by default.
```
If and when we merge the two and deprecate SizeTieredCompactionStrategy (or make it an alias for ICS)
we shouldn't reference STCS anymore, but rather provide full, self-sufficient documentation for ICS.

raphaelsc replied:
agreed.

bhalevy commented on docs/architecture/compaction/compaction-strategies.rst:
```diff
@@ -70,7 +70,38 @@ Set the parameters for :ref:`Leveled Compaction <leveled-compaction-strategy-lcs
+Namely:
+
+* Continuously modifying existing rows results in each row being split across several SSTables, making reads slow, which doesn’t happen in Leveled compaction.
+* Obsolete data (overwritten or deleted columns) may accumulate across tiers, wasting space, for a long time, until it is finally merged. This can be mitigated by running major compaction from time to time.
```
and by using Space Amplification Goal to encourage cross-tier compaction of the two largest tiers earlier, as data accumulates in the second-largest tier.

raphaelsc replied:
yes, looks like we forgot to update this when SAG was introduced.

bhalevy commented on docs/cql/compaction.rst:
```diff
@@ -170,7 +170,106 @@ LCS options
+
+.. note:: ICS is only available for Scylla Enterprise customers
+
+When using ICS, SSTable runs are put in different buckets depending on their size.
```
s/put in/organized by/

bhalevy commented on docs/cql/compaction.rst:
```diff
@@ -170,7 +170,106 @@ LCS options
+
+``space_amplification_goal`` (default: null)
+
+:label-tip:`ScyllaDB Enterprise`
```
Do we still need this?

Cc @annastuchlik

tchaikov replied:
created #22175 to track this.

bhalevy commented on docs/getting-started/system-requirements.rst:
```diff
@@ -99,7 +99,7 @@ Time-window Compaction Strategy (TWCS)  50%          70%
 ======================================  ===========  ============
 
-Use the default ICS  unless you'll have a clear understanding that another strategy is better for your use case. More on :doc:`choosing a Compaction Strategy </architecture/compaction/compaction-strategies>`.
+Use the default ICS (ScyllaDB Enterprise) or STCS  (ScyllaDB Open Source) unless you'll have a clear understanding that another strategy is better for your use case. More on :doc:`choosing a Compaction Strategy </architecture/compaction/compaction-strategies>`.
```
outdated.

bhalevy commented on docs/operating-scylla/procedures/tips/production-readiness.rst:
```diff
@@ -21,6 +21,18 @@ Choose a Compaction Strategy
 ============================
 
 Each workload may require a specific strategy. Refer to :doc:`Choose a Compaction Strategy </architecture/compaction/compaction-strategies>` for details.
+If you are using Scylla Enterprise, in all cases ICS is preferred over STCS.
```
outdated.

bhalevy commented on docs/operating-scylla/procedures/tips/production-readiness.rst:
```diff
@@ -21,6 +21,18 @@ Choose a Compaction Strategy
+Incremental Compaction Strategy (ICS)
+.....................................
+
+If you are using Scylla Enterprise, we highly recommend using ICS (the default setting) for any table that you have, or considering to have STCS as a compaction strategy.
```
outdated.

raphaelsc commented:
> > tests fail because scylla-dtest in OSS expects STCS, I suggested we patch it so it accepts either temporarily.
> 
> That's fine. Please go ahead and submit a patch for scylla-dtest.

Submitted: https://github.com/scylladb/scylla-dtest/pull/5443

avikivity commented:
Please write this as a normal feature introduction. Do mention it's a port from enterprise, but the patch should be understood on its own: what incremental compaction strategy is, when the benefits are, how it's based on runs.

raphaelsc commented:
v2:
- rebased
- changed patch description to look like a new feature, rather than port.
- docs: updated sections indicating ICS is enterprise only.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14184/)
:x: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14184/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: boost/incremental_compaction_test
:small_blue_diamond: boost/schema_change_test
:small_blue_diamond: boost/sstable_compaction_test
#### Build Details:
- Duration: 39 min
- Builder: spider1.cloudius-systems.com

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14187/)
#### Build Details:
- Duration: 27 min
- Builder: spider4.cloudius-systems.com

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14188/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14188/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: boost/incremental_compaction_test
:small_blue_diamond: boost/schema_change_test
:small_blue_diamond: boost/sstable_compaction_test
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/366/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5702/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13715/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2589/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14188/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/962/)
#### Build Details:
- Duration: 4 hr 54 min
- Builder: spider4.cloudius-systems.com

raphaelsc commented:
@scylladb/scylla-maint please consider merging

Surrina-ki commented:
@raphaelsc I don't see any particular dependencies with this. Can't it be used in the old version?

denesb commented:
> @raphaelsc I don't see any particular dependencies with this. Can't it be used in the old version?

Which old version? Previous Open-Source Scylla versions? It cannot be used in those because of the licensing.
